debezium-core/src/main/java/io/debezium/rest/MetricsResource.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.rest;
+
+import java.util.List;
+
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+
+import org.apache.kafka.connect.connector.Connector;
+import org.json.simple.JSONObject;
+
+import io.debezium.rest.jolokia.JolokiaAttributes;
+import io.debezium.rest.jolokia.JolokiaClient;
+
+public interface MetricsResource {
+    Connector getConnector();
+
+    String getAttributesFilePath();
```

Comment:
Is this really necessary? Do we know for what connector we are looking for metrics? If we do then it is just possible to set the pattern the attribute name files have to follow and just calculate it.
Also is it possible to have a common file with all attributes shared by all connectors and a per-connector definition?

Reply:
Is it even necessary to have the definition done in files? Would it be technically possible to re-use `io.debezium.metadata.ConnectorMetadata` for it?

Reply:
`ConnectorMetadata` class is used for the SchemaGenerator. I would personally not cross concerns with the REST extensions. Furthermore I think it is a good idea to have the implementation like this. That way each connector can freely decide how it provides the metrics definitions. Also mind in the future the REST extension needs to provide different versions of Debezium connectors. Using something hardcoded (outside the connector modules) is imo not a good idea. That way the connector can freely decide where and how metrics attributes are returned.
But maybe we could return something more "explicitly typed" here? Like having a helper method somewhere that loads the list of attributes from a file, but this method `getAttributesFilePath()`  could return a `List<String>` or `List<MetricsAttribute>` or even a new `MetricsAttributes` instance and let each connector give multiple options how this is created (from that txt files like now, a CONST, etc). wdyt @ani-sha  @jpechane ?

Reply:
I'd still like to see union of common config file and per-connector file as very large number of metrics is shared.

Reply:
And again if we come back to my original comment, do we really need it if the pattern is the same so the path can be just calculated?

Reply:
Do you think providing this flexibility does any harm? Also think about 3rd party connectors or community connectors that should eventually add up to the Debezium UI feature set. They might require flexibility.